src/libs/Middleware/SaveResponseInOnyx.js

```diff
@@ -14,50 +14,54 @@ const requestsToIgnoreLastUpdateID = ['OpenApp', 'ReconnectApp', 'GetMissingOnyx
+                return;
             }
+            const onyxUpdates = response.onyxData;
 
-            MemoryOnlyKeys.enable();
-            return true;
-        });
+            // Sometimes we call requests that are successfull but they don't have any response or any success/failure data to set. Let's return early since
+            // we don't need to store anything here.
+            if (!onyxUpdates && !request.successData && !request.failureData) {
+                return Promise.resolve(response);
+            }
+
+            // If there is an OnyxUpdate for using memory only keys, enable them
+            _.find(onyxUpdates, ({key, value}) => {
+                if (key !== ONYXKEYS.IS_USING_MEMORY_ONLY_KEYS || !value) {
+                    return false;
+                }
+
+                MemoryOnlyKeys.enable();
+                return true;
+            });
+
+            const responseToApply = {
+                type: CONST.ONYX_UPDATE_TYPES.HTTPS,
+                lastUpdateID: Number(response.lastUpdateID || 0),
+                previousUpdateID: Number(response.previousUpdateID || 0),
+                request,
+                response,
+            };
+
+            if (_.includes(requestsToIgnoreLastUpdateID, request.command) || !OnyxUpdates.doesClientNeedToBeUpdated(Number(response.previousUpdateID || 0))) {
+                return OnyxUpdates.apply(responseToApply);
+            }
+
+            // Save the update IDs to Onyx so they can be used to fetch incremental updates if the client gets out of sync from the server
+            OnyxUpdates.saveUpdateInformation(responseToApply);
 
-        const responseToApply = {
-            type: CONST.ONYX_UPDATE_TYPES.HTTPS,
-            lastUpdateID: Number(response.lastUpdateID || 0),
-            previousUpdateID: Number(response.previousUpdateID || 0),
-            request,
-            response,
-        };
-
-        if (_.includes(requestsToIgnoreLastUpdateID, request.command) || !OnyxUpdates.doesClientNeedToBeUpdated(Number(response.previousUpdateID || 0))) {
-            return OnyxUpdates.apply(responseToApply);
-        }
-
-        // Save the update IDs to Onyx so they can be used to fetch incremental updates if the client gets out of sync from the server
-        OnyxUpdates.saveUpdateInformation(responseToApply);
-
-        // Ensure the queue is paused while the client resolves the gap in onyx updates so that updates are guaranteed to happen in a specific order.
-        return Promise.resolve({
-            ...response,
-            shouldPauseQueue: true,
+            // Ensure the queue is paused while the client resolves the gap in onyx updates so that updates are guaranteed to happen in a specific order.
+            return Promise.resolve({
+                ...response,
+                shouldPauseQueue: true,
+            });
+        })
+        .catch((err) => {
+            console.error('Got exception while saving response in Onyx', err);
```

Comment:
This `catch` causes a regression, it suppresses network errors such as 'Failed to fetch', but I believe it shouldn't.

Unit tests defined in `APITest.js` are not failing due to a TypeError which is always raised in `SequentialQueue` file [here](https://github.com/Expensify/App/blob/5fcdfd9ebd187c11cab05c826ea10afec354b2fc/src/libs/Network/SequentialQueue.js#L74):

First an error is suppressed in `SaveResponseInOnyx`:
```ts
  console.error
    Got exception while saving response in Onyx Error: Failed to fetch
        at /tests/unit/APITest.js:90:40

      52 |         })
      53 |         .catch((err) => {
    > 54 |             console.error('Got exception while saving response in Onyx', err);
         |                     ^
      55 |         });
      56 |
      57 | export default SaveResponseInOnyx;
```

Later an error is raised as `shouldPauseQueue` isn't defined on `undefined` response:
```ts
    error1 =  TypeError: Cannot read properties of undefined (reading 'shouldPauseQueue')
        at /src/libs/Network/SequentialQueue.ts:73:26)

      at log (src/libs/Network/SequentialQueue.ts:81:21)
```

cc @mountiny @tienifr

Reply:
Thanks!